spring-integration-ip/src/test/java/org/springframework/integration/ip/tcp/TcpInboundGatewayTests.java

```diff
@@ -284,6 +291,57 @@ public void testErrorFlow() throws Exception {
 		scf.stop();
 	}
 
+	@Test
+	public void testNetCloseStream() throws InterruptedException, IOException {
+		testCloseStream(new TcpNetServerConnectionFactory(0),
+				port -> new TcpNetClientConnectionFactory("localhost", port));
+	}
+
+	@Test
+	public void testNioCloseStream() throws InterruptedException, IOException {
+		testCloseStream(new TcpNioServerConnectionFactory(0),
+				port -> new TcpNioClientConnectionFactory("localhost", port));
+	}
+
+	private void testCloseStream(AbstractServerConnectionFactory scf,
+			Function<Integer, AbstractClientConnectionFactory> ccf) throws InterruptedException, IOException {
+
+		scf.setSingleUse(true);
+		scf.setDeserializer(new ByteArrayRawSerializer());
+		TcpInboundGateway gateway = new TcpInboundGateway();
+		gateway.setConnectionFactory(scf);
+		BeanFactory bf = mock(ConfigurableBeanFactory.class);
+		gateway.setBeanFactory(bf);
+		gateway.start();
+		TestingUtilities.waitListening(scf, 20000L);
+		int port = scf.getPort();
+		final DirectChannel channel = new DirectChannel();
+		gateway.setRequestChannel(channel);
+		BridgeHandler bridge = new BridgeHandler();
+		bridge.setBeanFactory(bf);
+		bridge.afterPropertiesSet();
+		ConsumerEndpointFactoryBean consumer = new ConsumerEndpointFactoryBean();
+		consumer.setInputChannel(channel);
+		consumer.setBeanFactory(bf);
+		consumer.setHandler(bridge);
+		consumer.afterPropertiesSet();
+		consumer.start();
+		AbstractClientConnectionFactory client = ccf.apply(port);
+		CountDownLatch latch = new CountDownLatch(1);
+		client.registerListener(message -> {
+			latch.countDown();
+			return false;
+		});
+		client.afterPropertiesSet();
+		client.start();
+		TcpConnectionSupport connection = client.getConnection();
+		connection.send(new GenericMessage<>("foo"));
+		connection.shutdownOutput(); // signal EOF to server
+		assertThat(latch.await(10, TimeUnit.SECONDS)).isTrue();
+		gateway.stop();
+		client.stop();
```

Comment:
Don't we need to destroy `scf` in the end as well?

Reply:
No; the gateway stops the scf (and it's not a `DisposableBean`).